app/policies/macro_policy.rb

```diff
@@ -8,15 +8,17 @@ def create?
   end
 
   def show?
+    return @record.global? if !author? && @account_user.agent?
+
     true
   end
 
   def update?
-    true
+    author? || @account_user.administrator?
   end
 
   def destroy?
-    true
+    author? || orphan_record?
```

Comment:
@tejaswinichile  should we let agents delete orphan macros ? I think it makes more sense to give this permission only to admins.

Reply:
Yes, that does make sense but if the admin is clogged up with the work then we should let the agent delete these records. May be in the future we can come up with the archived state for the orphan records and then let the agent archive it. @pranavrajs Can you confirm?

Reply:
Let us not keep the private macros for the admin to delete them. When an agent is deleted, all their private macros are deleted and public macros would get orphaned. Then, all the orphan records can be deleted by an admin. I hope this would solve the burden of an admin to delete everything. Public Macros are anyways helpful for most of the users, so it makes sense to keep them and giving control to an admin to update it.